### ISSUE TITLE
[Bug]: Proxy pod keep restarting with error `find no available querycoord, check querycoord state` after etcd pod kill chaos test

### ISSUE THREAD

**Issue by zhuwenxing** [milvus-io/milvus#25391]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:2.2.0-20230704-3e055a52
- Deployment mode(standalone or cluster):cluster
- MQ type(rocksmq, pulsar or kafka):kafka    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

```
[2023-07-06T00:56:54.582Z] + kubectl get pods -o wide

[2023-07-06T00:56:54.582Z] + grep etcd-pod-kill-5751

[2023-07-06T00:56:54.582Z] etcd-pod-kill-5751-0                                              1/1     Running            0                 12m     10.104.9.162    4am-node14   <none>           <none>

[2023-07-06T00:56:54.582Z] etcd-pod-kill-5751-1                                              1/1     Running            0                 11m     10.104.1.65     4am-node10   <none>           <none>

[2023-07-06T00:56:54.582Z] etcd-pod-kill-5751-2                                              1/1     Running            0                 11m     10.104.21.59    4am-node24   <none>           <none>

[2023-07-06T00:56:54.582Z] etcd-pod-kill-5751-kafka-0                                        2/2     Running            2 (17m ago)       18m     10.104.1.50     4am-node10   <none>           <none>

[2023-07-06T00:56:54.582Z] etcd-pod-kill-5751-kafka-1                                        2/2     Running            3 (17m ago)       18m     10.104.9.155    4am-node14   <none>           <none>

[2023-07-06T00:56:54.582Z] etcd-pod-kill-5751-kafka-2                                        2/2     Running            2 (17m ago)       18m     10.104.6.130    4am-node13   <none>           <none>

[2023-07-06T00:56:54.582Z] etcd-pod-kill-5751-kafka-exporter-7954f8b6dd-spdqz                1/1     Running            4 (17m ago)       18m     10.104.1.39     4am-node10   <none>           <none>

[2023-07-06T00:56:54.582Z] etcd-pod-kill-5751-milvus-datacoord-fbc647bc9-b2tz8               1/1     Running            1 (10m ago)       18m     10.104.1.36     4am-node10   <none>           <none>

[2023-07-06T00:56:54.582Z] etcd-pod-kill-5751-milvus-datanode-5989ffb55b-2vh6h               1/1     Running            1 (11m ago)       18m     10.104.1.38     4am-node10   <none>           <none>

[2023-07-06T00:56:54.582Z] etcd-pod-kill-5751-milvus-datanode-5989ffb55b-qznqg               1/1     Running            1 (11m ago)       18m     10.104.9.142    4am-node14   <none>           <none>

[2023-07-06T00:56:54.582Z] etcd-pod-kill-5751-milvus-indexcoord-6dccf4bd9b-mfgqk             1/1     Running            1 (10m ago)       18m     10.104.1.32     4am-node10   <none>           <none>

[2023-07-06T00:56:54.582Z] etcd-pod-kill-5751-milvus-indexnode-675c656855-wzr9s              1/1     Running            1 (11m ago)       18m     10.104.6.123    4am-node13   <none>           <none>

[2023-07-06T00:56:54.582Z] etcd-pod-kill-5751-milvus-proxy-5dfdd8c494-jfx26                  0/1     Running            4 (6s ago)        18m     10.104.9.147    4am-node14   <none>           <none>

[2023-07-06T00:56:54.582Z] etcd-pod-kill-5751-milvus-querycoord-68c9bcd7dc-wwm4g             1/1     Running            0                 18m     10.104.9.146    4am-node14   <none>           <none>

[2023-07-06T00:56:54.582Z] etcd-pod-kill-5751-milvus-querynode-f6bdf7485-ctgzz               1/1     Running            1 (10m ago)       18m     10.104.1.33     4am-node10   <none>           <none>

[2023-07-06T00:56:54.582Z] etcd-pod-kill-5751-milvus-querynode-f6bdf7485-dvwng               1/1     Running            1 (10m ago)       18m     10.104.6.124    4am-node13   <none>           <none>

[2023-07-06T00:56:54.582Z] etcd-pod-kill-5751-milvus-querynode-f6bdf7485-qdqsr               1/1     Running            1 (10m ago)       18m     10.104.9.145    4am-node14   <none>           <none>

[2023-07-06T00:56:54.582Z] etcd-pod-kill-5751-milvus-rootcoord-7c586bd4c9-v8qfl              1/1     Running            1 (10m ago)       18m     10.104.1.34     4am-node10   <none>           <none>

[2023-07-06T00:56:54.582Z] etcd-pod-kill-5751-minio-0                                        1/1     Running            0                 18m     10.104.1.49     4am-node10   <none>           <none>

[2023-07-06T00:56:54.582Z] etcd-pod-kill-5751-minio-1                                        1/1     Running            0                 18m     10.104.9.157    4am-node14   <none>           <none>

[2023-07-06T00:56:54.582Z] etcd-pod-kill-5751-minio-2                                        1/1     Running            0                 18m     10.104.6.129    4am-node13   <none>           <none>

[2023-07-06T00:56:54.582Z] etcd-pod-kill-5751-minio-3                                        1/1     Running            0                 18m     10.104.20.111   4am-node22   <none>           <none>

[2023-07-06T00:56:54.582Z] etcd-pod-kill-5751-zookeeper-0                                    1/1     Running            0                 18m     10.104.9.152    4am-node14   <none>           <none>

[2023-07-06T00:56:54.582Z] etcd-pod-kill-5751-zookeeper-1                                    1/1     Running            0                 18m     10.104.6.134    4am-node13   <none>           <none>

[2023-07-06T00:56:54.582Z] etcd-pod-kill-5751-zookeeper-2                                    1/1     Running            0                 18m     10.104.1.51     4am-node10   <none>           <none>
```

`[2023-07-06T00:56:54.582Z] etcd-pod-kill-5751-milvus-proxy-5dfdd8c494-jfx26                  0/1     Running            4 (6s ago)        18m     10.104.9.147    4am-node14   <none>           <none>`

```
[2023/07/06 00:57:36.989 +00:00] [WARN] [client/client.go:94] ["QueryCoordClient msess key not existed"] [key=querycoord]
[2023/07/06 00:57:36.989 +00:00] [ERROR] [grpcclient/client.go:161] ["failed to get client address"] [error="find no available querycoord, check querycoord state"] [stack="github.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase[...]).connect\n\t/go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:161\ngithub.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase[...]).GetGrpcClient\n\t/go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:134\ngithub.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase[...]).callOnce\n\t/go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:271\ngithub.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase[...]).ReCall\n\t/go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:350\ngithub.com/milvus-io/milvus/internal/distributed/querycoord/client.(*Client).GetComponentStates\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/querycoord/client/client.go:121\ngithub.com/milvus-io/milvus/internal/util/funcutil.WaitForComponentStates.func1\n\t/go/src/github.com/milvus-io/milvus/internal/util/funcutil/func.go:67\ngithub.com/milvus-io/milvus/internal/util/retry.Do\n\t/go/src/github.com/milvus-io/milvus/internal/util/retry/retry.go:42\ngithub.com/milvus-io/milvus/internal/util/funcutil.WaitForComponentStates\n\t/go/src/github.com/milvus-io/milvus/internal/util/funcutil/func.go:91\ngithub.com/milvus-io/milvus/internal/util/funcutil.WaitForComponentHealthy\n\t/go/src/github.com/milvus-io/milvus/internal/util/funcutil/func.go:106\ngithub.com/milvus-io/milvus/internal/distributed/proxy.(*Server).init\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/proxy/service.go:470\ngithub.com/milvus-io/milvus/internal/distributed/proxy.(*Server).Run\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/proxy/service.go:298\ngithub.com/milvus-io/milvus/cmd/components.(*Proxy).Run\n\t/go/src/github.com/milvus-io/milvus/cmd/components/proxy.go:51\ngithub.com/milvus-io/milvus/cmd/roles.runComponent[...].func1\n\t/go/src/github.com/milvus-io/milvus/cmd/roles/roles.go:120"]
[2023/07/06 00:57:36.989 +00:00] [ERROR] [grpcclient/client.go:353] ["ClientBase ReCall grpc second call get error"] [role=querycoord] [error="err: find no available querycoord, check querycoord state\n, /go/src/github.com/milvus-io/milvus/internal/util/trace/stack_trace.go:51 github.com/milvus-io/milvus/internal/util/trace.StackTrace\n/go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:352 github.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase[...]).ReCall\n/go/src/github.com/milvus-io/milvus/internal/distributed/querycoord/client/client.go:121 github.com/milvus-io/milvus/internal/distributed/querycoord/client.(*Client).GetComponentStates\n/go/src/github.com/milvus-io/milvus/internal/util/funcutil/func.go:67 github.com/milvus-io/milvus/internal/util/funcutil.WaitForComponentStates.func1\n/go/src/github.com/milvus-io/milvus/internal/util/retry/retry.go:42 github.com/milvus-io/milvus/internal/util/retry.Do\n/go/src/github.com/milvus-io/milvus/internal/util/funcutil/func.go:91 github.com/milvus-io/milvus/internal/util/funcutil.WaitForComponentStates\n/go/src/github.com/milvus-io/milvus/internal/util/funcutil/func.go:106 github.com/milvus-io/milvus/internal/util/funcutil.WaitForComponentHealthy\n/go/src/github.com/milvus-io/milvus/internal/distributed/proxy/service.go:470 github.com/milvus-io/milvus/internal/distributed/proxy.(*Server).init\n/go/src/github.com/milvus-io/milvus/internal/distributed/proxy/service.go:298 github.com/milvus-io/milvus/internal/distributed/proxy.(*Server).Run\n/go/src/github.com/milvus-io/milvus/cmd/components/proxy.go:51 github.com/milvus-io/milvus/cmd/components.(*Proxy).Run\n"] [stack="github.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase[...]).ReCall\n\t/go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:353\ngithub.com/milvus-io/milvus/internal/distributed/querycoord/client.(*Client).GetComponentStates\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/querycoord/client/client.go:121\ngithub.com/milvus-io/milvus/internal/util/funcutil.WaitForComponentStates.func1\n\t/go/src/github.com/milvus-io/milvus/internal/util/funcutil/func.go:67\ngithub.com/milvus-io/milvus/internal/util/retry.Do\n\t/go/src/github.com/milvus-io/milvus/internal/util/retry/retry.go:42\ngithub.com/milvus-io/milvus/internal/util/funcutil.WaitForComponentStates\n\t/go/src/github.com/milvus-io/milvus/internal/util/funcutil/func.go:91\ngithub.com/milvus-io/milvus/internal/util/funcutil.WaitForComponentHealthy\n\t/go/src/github.com/milvus-io/milvus/internal/util/funcutil/func.go:106\ngithub.com/milvus-io/milvus/internal/distributed/proxy.(*Server).init\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/proxy/service.go:470\ngithub.com/milvus-io/milvus/internal/distributed/proxy.(*Server).Run\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/proxy/service.go:298\ngithub.com/milvus-io/milvus/cmd/components.(*Proxy).Run\n\t/go/src/github.com/milvus-io/milvus/cmd/components/proxy.go:51\ngithub.com/milvus-io/milvus/cmd/roles.runComponent[...].func1\n\t/go/src/github.com/milvus-io/milvus/cmd/roles/roles.go:120"]

```

### Milvus Log

failed job: https://qa-jenkins.milvus.io/blue/organizations/jenkins/chaos-test-kafka-for-release-cron/detail/chaos-test-kafka-for-release-cron/5751/pipeline

log:

[artifacts-etcd-pod-kill-5751-server-logs (2).tar.gz](https://github.com/milvus-io/milvus/files/11974026/artifacts-etcd-pod-kill-5751-server-logs.2.tar.gz)

[artifacts-etcd-pod-kill-5751-pytest-logs.tar.gz](https://github.com/milvus-io/milvus/files/11974027/artifacts-etcd-pod-kill-5751-pytest-logs.tar.gz)

**Comment by yanliang567** [comment]
@yah01 any updates

**Comment by yangle9567** [comment]
请问这是没人解决过吗？

**Comment by yanliang567** [comment]
this was fixed in milvus 2.3.x, please try on latest release v2.3.6
